subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalJavaCompileIntegrationTest.groovy

```diff
@@ -123,6 +124,41 @@ class IncrementalJavaCompileIntegrationTest extends AbstractIntegrationSpec impl
         executedAndNotSkipped(libraryCompileJava)
     }
 
+    @Unroll
+    def "does not recompile when only compileOptions.incremental property changes from #from to #to"() {
```

Comment:
I am not so sure where to place this test. It is actually not really related to incremental Java compile - it is actually related to incremental build. Maybe `JavaCompileIntegrationTest` is the way to go?

Reply:
It _is_ about enabling/disabling the incremental compiler, so I'd leave it here.